docs/assets/diagrams/transfer/03-confirm-payee.puml

```diff
@@ -0,0 +1,118 @@
+@startuml
+
+title Confirm Payee
+
+hide footbox
+
+autonumber
+
+box "Mobile Device" #Light
+  actor Alice
+  participant "PISP App" as App
+end box
+box "PISP" #Light
+  database Firebase
+  participant "PISP Server" as Server
+end box
+box "Mojaloop" #Light
+  participant "Switch" as Mojaloop
+end box
+
+Alice -> App: Initiate this transaction
+
+rnote right of App #Light
+  ""firebase.firestore()""
+  ""  .collection("transactions")""
+  ""  .doc("456")""
+  ""  .set({""
+  ""    sourceAccountId: "bob.fspb",""
+  ""    consentId: "123",""
+  ""    amount: {""
+  ""      amount: "20",""
+  ""      currency: "USD"""
+  ""    },""
+  ""    payer: {""
+  ""      personalInfo: {""
+  ""        complexName: {""
+  ""          firstName: "Bob",""
+  ""          lastName: "Beaver"""
+  ""        }""
+  ""      },""
+  ""      partyIdInfo: {""
+  ""        partyIdType: "MSISDN",""
+  ""        partyIdentifier: "+1-111-111-1111",""
+  ""        fspId: "fspb",""
+  ""      }""
+  ""    },""
+  ""  }, { merge: true })""
+end rnote
+
+App -> Firebase ++: Create a new transaction document
+Firebase <-- App: ""200 OK""
+
+rnote right of Firebase #Light
+  ""event(""
+  ""  "document.create",""
+  ""  "transactions/456",""
+  ""  <snapshot object>""
+  "")""
+end rnote
+
+Firebase -> Server ++: Notify creation of transaction document
+Firebase <-- Server: ""200 OK""
+deactivate Firebase
+
+rnote right of Server #Light 
+  ""**POST /thirdpartyRequests/transactions**""
+  ""FSPIOP-Source: pisp""
+  ""FSPIOP-Destination: fspb""
+  ""{""
+  ""  "transactionRequestId": "456",""
+  ""  "sourceAccountId": "bob.fspb",""
+  ""  "consentId": "789",""
+  ""  "payee": {""
+  ""    "partyIdInfo": {""
+  ""      "partyIdType": "MSISDN",""
+  ""      "partyIdentifier": "+1-222-222-2222",""
+  ""      "fspId": "fspa"""
+  ""    }""
+  ""  },""
+  ""  "payer": {""
```

Comment:
Does the app need to specify this or should/can it be done by the server?

Reply:
In the current design, the app need to specify the payer field, because otherwise the server would need to make a query to Firebase for every transaction document. I think the data will be stored in Firebase after once an account has been linked, but the app is also expected to cache the value (e.g, after linking an account or when the app is opened for the first time).

Reply:
So what happens if the app is uninstalled and installed back again?

Also can you elaborate on "I think the data will be stored in Firebase after once an account has been linked"

Reply:
When the app runs for the first time, it will go through the following process:
1. Query to the server about what are the bank accounts that have been linked with a particular PISP account.
2. All the linked bank accounts are shown as "session expired" since we know that they have been linked previously but there's no consent to make transactions stored in the device.
3. If a user wants to make a transaction, he/she must perform an account linking again to get `consentId`.
4. Upon performing the account linking, the app will get the party information for each linked bank account, which then will be put in the `payer` field when initiating a transaction.

Reply:
For point 1, how does the PISP server know what are the accounts has been linked? Do we persist that in firebase? I think this is related to your last point "that there could be a document to store what are the list of bank accounts that have been linked with a particular PISP account". You might want to make that decision sooner than later.

Reply:
Yeah, I think we can persist that ... but haven't really thought about it since we haven't discussed the linking process in detail...

For now, can we assume that if an app is uninstalled and then installed again, it will be considered as a fresh start (no account has been linked)? This is based on the assumption that all the public-private key pairs will be gone if the app is uninstalled, so we always have to perform an account linking before making any transaction.

Reply:
Sure. But it doesn't sound like we have enough information to do what point 2 is saying as in displaying "session expired". I think it's fine just to go back to a blank slate for now just to make things easier (and manage less state).

Reply:
Yeah, I think we can discuss about this further when deciding on the linking flow